Plugins/Json/include/Acts/Plugins/Json/JsonHierarchicalObjectConverter.hpp

```diff
@@ -0,0 +1,183 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2017-2019 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <Acts/Geometry/TrackingVolume.hpp>
+#include <Acts/Surfaces/Surface.hpp>
+#include "Acts/Geometry/HierarchicalGeometryContainer.hpp"
+#include "Acts/Geometry/TrackingGeometry.hpp"
+#include "Acts/Utilities/Definitions.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <nlohmann/json.hpp>
+
+#include <map>
+
+namespace Acts {
+
+/// @class JsonHierarchicalObjectConverter
+///
+/// @brief Convert Hierarchical Object Container to Json file and vice-versa
+template <typename object_t>
+class JsonHierarchicalObjectConverter {
+ public:
+  /// @class Config
+  /// Configuration of the Reader/Writer
+  class Config {
+   public:
+    /// The detector tag
+    std::string detkey = "detector";
+    /// The volume identification string
+    std::string volkey = "volumes";
+    /// The boundary surface string
+    std::string boukey = "boundaries";
+    /// The layer identification string
+    std::string laykey = "layers";
+    /// The approach identification string
+    std::string appkey = "approach";
+    /// The sensitive identification string
+    std::string senkey = "sensitive";
+    /// The representing idntification string
+    std::string repkey = "representing";
+    /// The name identification
+    std::string namekey = "Name";
+    /// The object key
+    std::string datakey = "Object";
+
+    /// The default logger
+    std::shared_ptr<const Logger> logger;
+    /// The name of the writer
+    std::string name = "";
+
+    /// Steering to handle sensitive data
+    bool processSensitives = true;
+    /// Steering to handle approach data
+    bool processApproaches = true;
+    /// Steering to handle representing data
+    bool processRepresenting = true;
+    /// Steering to handle boundary data
+    bool processBoundaries = true;
+    /// Steering to handle volume data
+    bool processVolumes = true;
+
+    /// Constructor
+    ///
+    /// @param lname Name of the writer tool
+    /// @param lvl The output logging level
+    Config(const std::string& lname = "JsonHierarchicalObjectConverter",
+           Logging::Level lvl = Logging::INFO)
+        : logger(getDefaultLogger(lname, lvl)), name(lname) {}
+  };
+
+  using geo_id_value = uint64_t;
+  using Representation = std::map<geo_id_value, object_t>;
```

Comment:
This should use `GeometryID::Value` to avoid missmatches between the types. The `geo_id_value` typedef can be removed completely.

Reply:
Changed in last commit.